test/model/store_test.dart

```diff
@@ -652,6 +655,90 @@ void main() {
     test('retries on MalformedServerResponseException', () {
       checkRetry(() => connection.prepare(httpStatus: 200, body: 'nonsense'));
     });
+
+    group('reportErrorToUserBriefly', () {
+      String? lastReportedError;
+      String? takeLastReportedError() {
+        final result = lastReportedError;
+        lastReportedError = null;
+        return result;
+      }
+
+      /// This is an alternative to [ZulipApp]'s implementation of
+      /// [reportErrorToUserBriefly] for testing.
+      Future<void> logAndReportErrorToUserBriefly(String? message, {
+        String? details,
+      }) async {
+        if (message == null) return;
+        lastReportedError = '$message\n$details';
+      }
+
+      Future<void> prepare() async {
+        reportErrorToUserBriefly = logAndReportErrorToUserBriefly;
+        addTearDown(() => reportErrorToUserBriefly = defaultReportErrorToUserBriefly);
+
+        await prepareStore(lastEventId: 1);
+        updateMachine.debugPauseLoop();
+        updateMachine.poll();
+      }
+
+      late void Function() prepareError;
+
+      void pollAndFail(FakeAsync async) {
+        prepareError();
+        updateMachine.debugAdvanceLoop();
```

Comment:
```suggestion
      void pollAndFail(FakeAsync async) {
        updateMachine.debugAdvanceLoop();
```

Instead the callers can say `connection.prepare(…)` (i.e., the body of what they would set `prepareError` to) just before each call to `pollAndFail`. That isn't really any longer — it makes a very slight duplication of the content of those `connection.prepare` calls — and it makes the control flow substantially more straightforward.